kotlin-example/src/main/kotlin/org/onflow/examples/kotlin/streaming/streamEvents/SubscribeEventsExample.kt

```
@@ -0,0 +1,67 @@
+        receivedEvents: MutableList<FlowEvent>
+    ) {
+        val dataJob = scope.launch {
+            try {
+                for (events in dataChannel) {
+                    if (!isActive) break
+                    if (events.isNotEmpty()) {
+                        receivedEvents.addAll(events)
+                    }
+                    yield()  // Ensure coroutine checks for cancellation
+                }
+            } catch (e: CancellationException) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)